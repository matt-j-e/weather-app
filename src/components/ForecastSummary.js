import React from "react";
import PropTypes from "prop-types";
import WeatherIcon from "react-icons-weather";
import moment from "moment";

import "../styles/ForecastSummary.css";

const ForecastSummary = ({ forecast, onSelect }) => {
  const { date, description, icon, temperature } = forecast;
  return (
    <div
      className="forecast-summary"
      data-testid="forecast-summary"
      onClick={() => onSelect(date)}
      onKeyPress={() => onSelect(date)}
      role="button"
      tabIndex="0"
    >
      <div className="forecast-summary__date">{moment(date).format("ddd")}</div>
      <div className="forecast-summary__icon" data-testid="forecast-icon">
        <WeatherIcon name="owm" iconId={icon} />
      </div>
      <div className="forecast-summary__temperature">
        {temperature.max}&deg;c
      </div>
      <div className="forecast-summary__description">{description}</div>
      {/* <button type="button" onClick={() => onSelect(date)}>
        More details
      </button> */}
    </div>
  );
};

ForecastSummary.propTypes = {
  forecast: PropTypes.shape({
    date: PropTypes.number.isRequired,
    description: PropTypes.string.isRequired,
    icon: PropTypes.string.isRequired,
    temperature: PropTypes.shape({
      max: PropTypes.number,
      min: PropTypes.number,
    }).isRequired,
  }).isRequired,
  onSelect: PropTypes.func.isRequired,
};

export default ForecastSummary;
