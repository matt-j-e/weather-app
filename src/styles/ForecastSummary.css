.forecast-summary {
    flex-basis: 19%;
    font-size: 0.8rem;
    text-align: center;
    margin: 0.1rem;
    padding: 0.25rem 0;
    /* background-color: wheat; */
    outline: none;
}

.forecast-summary__date {
    font-weight: bold;
}

.forecast-summary__temperature, .forecast-summary__description {
    padding: 0.4rem;
}

.forecast-summary__icon .wi {
    font-size: 1.4rem;
    padding: 1rem;
}

button {
    margin-top: 1rem;
}

/*** MEDIA QUERIES ***/
@media screen and (min-width: 650px) {
    .forecast-summary {
        font-size: 150%;;
    }
}