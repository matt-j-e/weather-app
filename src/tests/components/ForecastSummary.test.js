import React from "react";
import { render } from "@testing-library/react";
import ForecastSummary from "../../components/ForecastSummary";

describe("ForecastSummary", () => {
  const validProps = {
    forecast: {
      date: 111,
      description: "Stub description",
      icon: "800",
      temperature: {
        max: 22,
        min: 11,
      },
    },
    onSelect: jest.fn(),
  };

  it("renders correctly", () => {
    const { asFragment } = render(
      <ForecastSummary
        forecast={validProps.forecast}
        onSelect={validProps.onSelect}
      />
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it("renders correct values for props", () => {
    const { getByText, getByTestId } = render(
      <ForecastSummary
        forecast={validProps.forecast}
        onSelect={validProps.onSelect}
      />
    );
    expect(getByText("Thu")).toHaveClass("forecast-summary__date");
    expect(getByText("Stub description")).toHaveClass(
      "forecast-summary__description"
    );
    expect(getByTestId("forecast-icon")).toHaveClass("forecast-summary__icon");
    expect(getByText("22Â°c")).toHaveClass("forecast-summary__temperature");
  });
});
