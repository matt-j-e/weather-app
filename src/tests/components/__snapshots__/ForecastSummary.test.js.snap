// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ForecastSummary renders correctly 1`] = `
<DocumentFragment>
  <div
    class="forecast-summary"
    data-testid="forecast-summary"
    role="button"
    tabindex="0"
  >
    <div
      class="forecast-summary__date"
    >
      Thu
    </div>
    <div
      class="forecast-summary__icon"
      data-testid="forecast-icon"
    >
      <i
        class="wi wi-day-sunny"
        name="owm"
      />
    </div>
    <div
      class="forecast-summary__temperature"
    >
      22Â°c
    </div>
    <div
      class="forecast-summary__description"
    >
      Stub description
    </div>
  </div>
</DocumentFragment>
`;
